tests/spl/token-proxy/programs/token-proxy/src/lib.rs

```diff
@@ -170,7 +170,7 @@ pub struct ProxyCreateAssociatedTokenAccount<'info> {
     #[account(mut)]
     pub authority: Signer<'info>,
     #[account(
-        init,
+        init_if_needed,
```

Comment:
It would be much better if you could add instructions and test cases that failed before your changes(cases in #2434) but now is successful instead of changing an existing test.

Reply:
Fyi, similar tests can be found at:
Associated token constraints - https://github.com/coral-xyz/anchor/blob/master/tests/misc/tests/misc/misc.ts#L904
`init_if_needed` - https://github.com/coral-xyz/anchor/blob/master/tests/misc/tests/misc/misc.ts#L1549

Reply:
Thanks Acheron, I'll jump back to this pr as soon as I can and add the specific tests.